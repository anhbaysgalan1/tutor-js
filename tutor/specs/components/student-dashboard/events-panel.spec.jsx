import React from 'react';
import SnapShot from 'react-test-renderer';
import { filter, map } from 'lodash';

import MOCK_DASHBOARD_RESPONSE from '../../../api/courses/1/dashboard';
import EventsPanel from '../../../src/components/student-dashboard/events-panel';

describe('EventsPanel', function() {

  beforeEach(function() {
    this.props = {
      events: MOCK_DASHBOARD_RESPONSE.tasks,
      courseId: '1',
      isCollege: false,
    };
  });

  it('renders with events', function() {
    expect(SnapShot.create(<EventsPanel {...this.props} />).toJSON()).toMatchSnapshot();
  });

  it('renders with events as named', function() {
    const wrapper = mount(<EventsPanel {...this.props} />);
    const renderedTitles = wrapper.find('.title').map(item => item.text());
    const mockTitles = map(MOCK_DASHBOARD_RESPONSE.tasks, 'title');
    expect(renderedTitles).to.deep.equal(mockTitles);
  });

  it('renders late only for homework when isCollege is false', function() {
    const wrapper = mount(<EventsPanel {...this.props} />);
    const mockHomeworkTasks = filter(
      MOCK_DASHBOARD_RESPONSE.tasks, { type: 'homework', complete: false }
    );
    expect(wrapper.find('.late').length).to.equal(mockHomeworkTasks.length);
  });

  it('renders late only for all tasks when isCollege is true', function() {
    this.props.isCollege = true;
    const wrapper = mount(<EventsPanel {...this.props} />);
    const mockTasks = filter(MOCK_DASHBOARD_RESPONSE.tasks, { complete: false });
    expect(wrapper.find('.late').length).to.equal(mockTasks.length);
  });
});
