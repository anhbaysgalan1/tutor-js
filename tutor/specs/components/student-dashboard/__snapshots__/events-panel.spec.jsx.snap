// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsPanel renders with events 1`] = `
<div
  className="panel panel-default"
  id={undefined}
>
  <div
    className="panel-body"
  >
    <div
      className="row labels"
    >
      <div
        className="col-sm-7 col-xs-12"
      >
        <span
          className="date-range"
        >
          <time>
            Oct 14, 2017
          </time>
          <span>
            â€“
          </span>
          <time>
            Oct 14, 2017
          </time>
        </span>
      </div>
      <div
        className="progress-label col-sm-3 col-sm-offset-0 col-xs-5 col-xs-offset-2"
      >
        Progress
      </div>
      <div
        className="due-at-label col-sm-2 col-xs-5"
      >
        Due
      </div>
    </div>
    <a
      aria-label="Work on homework: Chapter 6 Homework Before Due, no feedback"
      className="task row homework"
      data-event-id="6"
      href="/course/1/task/6"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="homework icon"
          className="icon icon-lg icon-homework"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        Chapter 6 Homework Before Due, no feedback
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          0/2 answered
        </span>
        <i
          className="info late"
          onBlur={[Function]}
          onClick={null}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 24, 7:45pm
        </time>
      </div>
    </a>
    <a
      aria-label="Work on reading: Chapter 5 and Chapter 6 Reading"
      className="task row reading"
      data-event-id="4"
      href="/course/1/task/4"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="reading icon"
          className="icon icon-lg icon-reading"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        Chapter 5 and Chapter 6 Reading
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          In progress
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 24, 7:45pm
        </time>
      </div>
    </a>
    <a
      aria-label="Work on reading: Chapter 10 and Chapter 11 Reading, multi-part"
      className="task row reading"
      data-event-id="9"
      href="/course/1/task/9"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="reading icon"
          className="icon icon-lg icon-reading"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        Chapter 10 and Chapter 11 Reading, multi-part
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          In progress
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 24, 7:45pm
        </time>
      </div>
    </a>
    <a
      aria-label="Work on reading: iReading 1: Force"
      className="task row reading"
      data-event-id="3"
      href="/course/1/task/3"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="reading icon"
          className="icon icon-lg icon-reading"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        iReading 1: Force
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          In progress
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 12, 3:15am
        </time>
      </div>
    </a>
    <a
      aria-label="Work on reading: iReading 2: Newton's First Law of Motion: Inertia"
      className="task row reading"
      data-event-id="5"
      href="/course/1/task/5"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="reading icon"
          className="icon icon-lg icon-reading"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        iReading 2: Newton's First Law of Motion: Inertia
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          Complete
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 13, 3:15am
        </time>
      </div>
    </a>
    <a
      aria-label="Work on reading: iReading 3: Newton's Second Law of Motion:"
      className="task row reading"
      data-event-id="7"
      href="/course/1/task/7"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="reading icon"
          className="icon icon-lg icon-reading"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        iReading 3: Newton's Second Law of Motion:
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          Not started
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 16, 9:15am
        </time>
      </div>
    </a>
    <a
      aria-label="Work on reading: iReading 4: Newton's Third Law of Motion"
      className="task row reading"
      data-event-id="10"
      href="/course/1/task/10"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="reading icon"
          className="icon icon-lg icon-reading"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        iReading 4: Newton's Third Law of Motion
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          In progress
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 19, 9:15am
        </time>
      </div>
    </a>
    <a
      aria-label="Work on homework: Homework #1"
      className="task row homework"
      data-event-id="11"
      href="/course/1/task/11"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="homework icon"
          className="icon icon-lg icon-homework"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        Homework #1
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          6/7 correct
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 20, 9:15am
        </time>
      </div>
    </a>
    <a
      aria-label="Work on homework: Homework #2"
      className="task row homework"
      data-event-id="13"
      href="/course/1/task/13"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="homework icon"
          className="icon icon-lg icon-homework"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        Homework #2
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          7/8 correct
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          Apr 21, 9:15am
        </time>
      </div>
    </a>
    <a
      aria-label="Work on homework: Homework #3"
      className="task row homework"
      data-event-id="15"
      href="/course/1/task/15"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="homework icon"
          className="icon icon-lg icon-homework"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        Homework #3
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          6/6 answered
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          May 04, 6:15pm
        </time>
      </div>
    </a>
    <a
      aria-label="Work on homework: Homework #4 (final)"
      className="task row homework"
      data-event-id="17"
      href="/course/1/task/17"
      onClick={[Function]}
      onKeyDown={undefined}
      tabIndex={-1}
    >
      <div
        className="column-icon col-sm-1 col-xs-2"
      >
        <i
          aria-label="homework icon"
          className="icon icon-lg icon-homework"
        />
      </div>
      <div
        className="title col-sm-6 col-xs-10"
      >
        Homework #4 (final)
      </div>
      <div
        className="feedback col-sm-3 col-xs-5"
      >
        <span>
          7/3 answered
        </span>
      </div>
      <div
        className="due-at col-sm-2 col-xs-5"
      >
        <time>
          May 19, 9:15am
        </time>
      </div>
    </a>
  </div>
</div>
`;
